pandas/io/excel/_base.py

```diff
@@ -289,6 +289,9 @@ def read_excel(io,
 
     if not isinstance(io, ExcelFile):
         io = ExcelFile(io, engine=engine)
+    elif engine and engine != io.engine:
```

Comment:
Is `io.engine` ever None? Coverage here might be a little suspect for cases like:

```python
read_excel(ExcelFile(), engine='xlrd')
```

which I suppose should work fine (assuming xlrd is default) but may not be covered here explicitly

Reply:
Current behaviour is `io.engine` in this line will default to 'xlrd' since that's in the ExcelFile constuctor -> added a test case to make this more explicit.